test/Compiler/simple/TypeSynonymArgs.agda

```diff
@@ -0,0 +1,35 @@
+module TypeSynonymArgs where
+
+postulate
+  Functor          : (Set → Set) → Set
+  Identity         : Set → Set
+  Identity-functor : Functor Identity
+
+{-# FOREIGN GHC import Data.Functor.Identity (Identity) #-}
+
+{-# FOREIGN GHC data FunctorDict f = Functor f => FunctorDict    #-}
+{-# COMPILE GHC Functor            = type(0) FunctorDict         #-} -- compiles to `type T_Functor_n = Functor`
+{-# COMPILE GHC Identity           = type(0) Identity            #-} -- compiles to `type T_Identity_m = Identity`
+{-# COMPILE GHC Identity-functor   = FunctorDict                 #-} -- `Identity` used without args (in type), errors when it is defined eta expanded
+
+--------
+
+postulate
+  X : Set → Set
+
+{-# FOREIGN GHC type HsSynonym a = a  #-}
+
+-- Equivalent
+{-# COMPILE GHC X = type(1) HsSynonym #-}
```

Comment:
This isn't very idiomatic Haskell syntax.  Could we have something like
```
COMPILE GHC X a = type HsSynonym a
```
instead?  This would be easier to understand.

Reply:
I think this would be backwards incompatible if zero arguments are specified like this `COMPILE GHC X = type HsSynonym`. Is Haskell-like syntax beneficial?

Reply:
> I think this would be backwards incompatible if zero arguments are specified like this `COMPILE GHC X = type HsSynonym`.

Could you elaborate the clash?

> Is Haskell-like syntax beneficial?

A priori yes, because then it is sort of self-explanatory.

Reply:
When writing arguments explicitly, how to express none arguments? Intuitively - by not writing anything. My first commit makes old syntax equivalent to zero arguments expanded and breaks when RHS is a type synonym with >0 required args, such as the builtin level polymorphic IO:

```
postulate
  IO : forall lvl -> Set lvl -> Set lvl

{-# FOREIGN GHC type AgdaIO lvl a = IO a #-}
{-# COMPILE GHC IO = type AgdaIO #-}
```

Reply:
Ok, I see the problem.  Old code would have to be updated to:
```
COMPILE GHC IO lvl a = type AgdaIO lvl a
```
Maybe this is the price to pay, if we want to give the user full control over eta-expansion for type synonyms.

Or, we make a new syntax for the case of forced arity zero, surpressing automatic eta-expansion:
```
COMPILE GHC Functor {} = type FunctorDict
```
The `{}` looks a bit weird, but I have no better idea for "don't insert arguments".

Other opinions, @nad @UlfNorell @jespercockx @gallais ?

Reply:
Would a lot of code break if we changed the syntax?

Reply:
Well, you would have to eta-expand next to all `{-# Compile GHC Foo = type Bar #-}` pragmas.